test.sh

```diff
@@ -3,33 +3,41 @@
 set -e
 
 # Stop all child processes on exit
-trap "pkill -P $$" EXIT
+trap cleanup EXIT
+
+cleanup() {
+    pkill -P $$
+    if [ -d "$tmpdir" ]; then
```

Comment:
I would feel better if this checked if it explicitly checked if `$tmpdir` was not empty, even though bash does appear to do the right thing (`-d` on an empty string returns false). 

```suggestion
    if [ ! -z "$tmpdir" ] && [ -d "$tmpdir" ]; then
```

I am very paranoid about commands that recursively delete directories.

Reply:
Better paranoid than careless! I'll add this.